rts/Game/Camera/RotOverheadController.h

```diff
@@ -19,16 +19,19 @@ class CRotOverheadController : public CCameraController
 	void MouseWheelMove(float move, const float3& newDir) { MouseWheelMove(move); }
 
 	void SetPos(const float3& newPos);
+	void SetRot(const float3& newRot) { rot = newRot; };
```

Comment:
Does this (and the same for the Spring camera below) mean that passing `rot` with a non-zero `z` component nets you a rolled (potentially upside-down etc) camera?

Reply:
Yes, this allows rotatable overhead camera to roll if the newRot has a `z` component. Unfortunately there is no mechanism to set this in practice, should I add a config value to enable a modifier to roll the camera? Perhaps MOVE_STATE_TLT. And/or set it with `SetCameraState`. 

As for switching to other cameras, the more constrained cameras manage their constraints to prevent rolls. Spring camera does it with `float3 GetRot() const { return (float3(rot.x, GetAzimuth(), 0.0f)); }`, so the current `rot.z` has no effect. The rest of the cameras don't override the CameraController SetRot, so they just set the `dir` and therefore cannot roll. Except for overview camera, which throws out the x and z components.

But in general, if a camera is expected to operate under certain constraints, then it should manage those constraints internally. For example, as overview camera does by throwing out or otherwise constraining the values. And we do see this all over, such as spring camera clamping the `rot.x` to the 90 degree range of .5pi to pi.

Reply:
Unfortunately camera rolling breaks a number of other things, so I won't be adding any way for the rotatable overhead camera to be rolled from gameside.